packages/playwright/src/fixture.ts

```diff
@@ -132,21 +133,140 @@ type Playwright = {
   _instrumentation: ClientInstrumentation;
 };
 
+let wsDeferred: Deferred<WebSocket> | undefined;
+
+async function waitUntilConnected() {
+  if (wsDeferred) {
+    return wsDeferred.promise;
+  }
+  const deferred = (wsDeferred = createDeferred());
+
+  const port = getServerPort();
+  const ws = new WebSocket(`ws://localhost:${port}`);
+
+  ws.on("open", () => deferred.resolve(ws));
+  ws.on("error", error => deferred.reject(error));
+  // TODO: close WS connections on test end
```

Comment:
This just got moved from the old place in the code. It will hopefully be removed soon because I'm working on ditching the WS-based communication altogether from this fixture.